### PR TITLE
[ANCHOR-590] Support wildcard `homeDomains` for sep10

### PR DESCRIPTION
**Pull request by JiahuiWho**
### Description

Introduce the capability to use a wildcard prefix in the SEP10_HOME_DOMAINS configuration. This enhancement will allow for more dynamic and flexible domain management, especially beneficial in multi-tenant settings.

### Context

1. Support wildcard match (e.g., *.stellar.org).

2. To Ensure backward compatibility, the existing configurations with specific domains (e.g. ap.stellar.org) continue to work seamlessly.

### Testing

- `./gradlew test`

### CODE REVIEW

**Comment by JakeUrban** [comment]
❓ should we support a wildcard prefix or just assume subdomains of domains provided are valid? For example, should anchor.stellar.org be valid if stellar.org was configured? Or should we require the business to explicitly configure a wildcard for subdomains?

File: core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java [hunk 1]
```diff
@@ -47,4 +50,34 @@ public static String fetchSigningKeyFromClientDomain(String clientDomain) throws
       throw new SepException(String.format("Invalid config: %s", e.getMessage()));
     }
   }
+
+  public static Boolean isDomainNameMatch(List<String> patternsAndDomains, String domainName) {
```
Comment (philipliu) [inline core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java:54]: Can we add Javadoc for these public methods?

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertySep10Config.java [hunk 2]
```diff
@@ -137,7 +137,9 @@ void validateConfig(Errors errors) {
             "sep10-web-auth-domain-invalid",
             "The sep10.web_auth_domain does not have valid format.");
       }
-    } else if (homeDomains != null && homeDomains.size() > 1) {
+    } else if (homeDomains != null
+        && !homeDomains.isEmpty()
+        && (homeDomains.size() > 1 || homeDomains.get(0).contains("*"))) {
```
Comment (philipliu) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertySep10Config.java:142]: Is the wildcard home domain always the first home domain in the list?

Comment (JiahuiWho) [reply platform/src/main/java/org/stellar/anchor/platform/config/PropertySep10Config.java:142]: No this is to make sure `web_auth_domain` must be set if
1.  `homeDomains` has more than one value
or  2.  `homeDomains` has only one value and it is a wildcard domain

File: core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java [hunk 3]
```diff
@@ -47,4 +50,34 @@ public static String fetchSigningKeyFromClientDomain(String clientDomain) throws
...
+        // Escape special characters in the pattern and replace '*' with '.*'
+        String regex = patternOrDomain.replace(".", "\\.").replace("*", ".*");
+        Pattern patternObject = Pattern.compile(regex);
+        Matcher matcher = patternObject.matcher(domainName);
+        if (matcher.matches()) {
+          return true;
+        }
+      } else {
+        // exact domain
+        if (patternOrDomain.equals(domainName)) {
```
[the remark this answers is not in the document]
Comment (JiahuiWho) [reply core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java:67]: Oh I separate the check for wildcard domain and exact domain .. so no pattern or matcher created for exact domain

**Review by philipliu — APPROVED** [review]
LGTM

**Pull request merged**